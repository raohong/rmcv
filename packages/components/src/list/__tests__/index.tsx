import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { List, ListLoadingStatus, listClassNames } from '..';

const testId = 'list';

it('render correctly', () => {
  const tree = render(<List />);

  expect(tree.asFragment()).toMatchSnapshot();
});

it('render with loadingStatus', () => {
  render(<List loadingStatus={ListLoadingStatus.LOADING} data-testid={testId} />);

  expect(
    screen.getByTestId(testId).querySelector(`.${listClassNames.loadingIcon}`),
  ).toBeInTheDocument();
});

it('render with renderLoading', () => {
  render(
    <List
      loadingStatus={ListLoadingStatus.LOADING}
      renderLoading={() => <span data-testid='loading' />}
    />,
  );

  expect(screen.getByTestId('loading')).toBeInTheDocument();
});

it('render with renderError', () => {
  render(
    <List
      loadingStatus={ListLoadingStatus.ERROR}
      renderError={() => <span data-testid='error' />}
    />,
  );

  expect(screen.getByTestId('error')).toBeInTheDocument();
});

it('render with renderFinished', () => {
  render(
    <List
      loadingStatus={ListLoadingStatus.FINISHED}
      renderFinished={() => <span data-testid='finished' />}
    />,
  );

  expect(screen.getByTestId('finished')).toBeInTheDocument();
});
