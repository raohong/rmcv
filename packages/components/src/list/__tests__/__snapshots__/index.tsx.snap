// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly 1`] = `
<DocumentFragment>
  .emotion-0 {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-tap-highlight-color: transparent;
  -webkit-font-smooth: antialiased;
  position: relative;
  touch-action: none;
  list-style: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,Helvetica Neue,Helvetica,Segoe UI,Arial,Roboto,PingFang SC,miui,Hiragino Sans GB,Microsoft Yahei,sans-serif;
}

.emotion-1 {
  box-sizing: border-box;
  color: #969799;
  font-size: 14px;
  padding: 16px 0;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

<div
    class="RVuiList-root emotion-0"
  >
    <div
      class="RVuiList-text emotion-1"
      tabindex="-1"
    />
  </div>
</DocumentFragment>
`;
